### PR TITLE
[libsocialcache] Add SyncHelper::syncProfiles property. Contributes t…

### PR DESCRIPTION
**Pull request by blammit**
…o JB#32794

### CODE REVIEW

File: src/qml/synchelper.cpp [hunk 1]
```diff
@@ -118,24 +114,56 @@ void SyncHelper::slotSyncStatus(const QString &aProfileId, int aStatus,
...
+    checkCurrentRun();
+}
+
+void SyncHelper::refreshSyncProfiles()
+{
+    if (!m_complete) {
+        return;
+    }
+    QStringList syncProfiles;
+    Q_FOREACH (const QString &profileId, m_interface->syncProfilesByType(QStringLiteral("sync"))) {
```
Comment (chriadam) [inline src/qml/synchelper.cpp]: while this will be the case for images sync, it may not be for other types (e.g, backup storage etc)

Comment (blammit) [inline src/qml/synchelper.cpp]: Thanks, fixed to call allVisibleSyncProfiles() instead

**Comment by blammit** [comment]
Actually maybe I can just call syncProfilesByType() with an empty string instead...

**Comment by blammit** [comment]
Actually it turns out the "type" is just used by syncProfilesByType() to figure out the directory that contains the sync profile, which should always be "sync" afaik.

**Comment by chriadam** [comment]
Oh!  I see, it's sync vs client profiles (Buteo concepts).  Makes sense.  LGTM.

**Comment by blammit** [comment]
Yep, confusing :/

**Pull request merged**